src/main.ts

```diff
@@ -214,6 +214,14 @@ function main(args: string[]): number {
     return 1;
   }
 
+  if (config.options.module !== ts.ModuleKind.CommonJS) {
+    // This is not an upstream TypeScript diagnostic, therefore it does not go
+    // through the diagnostics array mechanism.
+    console.error(
+        'tsickle converts TypeScript modules to Closure modules, via CommonJS internally. Set tsconfig.js "module": "commonjs"');
+    return 2;
```

Comment:
Also done, see commit. I have not the ground the source code of very many Google projects, there seems to be disagreement among experts as to whether it is best to consolidate to a single error exit return value, or to have as many different values as possible for different things that went wrong. I'm equally happy with either, I have found that scripts rarely care why command failed.